tests/specs/stn/with-replay-error-mode.e2e.js

```diff
@@ -0,0 +1,152 @@
+  ]
+})
+
+describe.withBrowsersMatching(notIE)('Trace error mode', () => {
+  let initSTReceived, getFirstSTPayload, getReplayOnErrorUrl
+  beforeEach(async () => {
+    await browser.destroyAgentSession()
+    await browser.testHandle.scheduleReply('bamServer', {
+      test: testRumRequest,
+      permanent: true,
+      body: JSON.stringify({ stn: 0, err: 1, ins: 1, spa: 1, sr: 1, loaded: 1 })
+    })
+
+    initSTReceived = undefined
+    getFirstSTPayload = browser.testHandle.expectResources().then(resPayload => initSTReceived = resPayload)
+    getReplayOnErrorUrl = browser.testHandle.assetURL('stn/instrumented.html', config({ session_replay: { sampleRate: 0, errorSampleRate: 1 }, session_trace: { harvestTimeSeconds: 2 } }))
+  })
+
+  function simulateErrorInBrowser () { // this is a way to throw error in WdIO / Selenium without killing the test itself
+    const errorElem = document.createElement('script')
+    errorElem.textContent = 'throw new Error("triggered! 0__0");'
+    document.body.append(errorElem)
+  }
+
+  it('switches to full mode when an error happens after page load', async () => {
+    await browser.url(await getReplayOnErrorUrl)
+    await browser.waitForAgentLoad()
+    await browser.waitUntil(() => browser.execute(async function () { // here for no-retry wdio stability
+      return await Object.values(newrelic.initializedAgents)[0].features.session_trace.onAggregateImported
+    }), { timeout: 5000 })
+    let startingMode = (await getTraceMode())[0]
+    expect(initSTReceived).toBeUndefined()
+    expect(startingMode).toEqual(MODE.ERROR)
+
+    await browser.execute(simulateErrorInBrowser)
+    await getFirstSTPayload
+    let modeAfterErr = (await getTraceMode())[0]
+    expect(modeAfterErr).toEqual(MODE.FULL)
+    // The loadEventEnd is part of PT and PNT resource entry and is a node created at start of page life.
+    expect(initSTReceived.request.body.res.find(node => node.n === 'loadEventEnd')).toEqual(expect.objectContaining({ n: 'loadEventEnd', o: 'document', t: 'timing' }))
+
+    await expect(browser.testHandle.expectResources(3000)).resolves.toBeTruthy() // double check there's nothing wrong with full mdoe interval harvest
+  })
+
+  it('starts in full mode when an error happens before page load', async () => {
+    await browser.url(await getReplayOnErrorUrl)
+    await browser.execute(simulateErrorInBrowser)
+    await browser.waitForAgentLoad()
+    await expect(getFirstSTPayload).resolves.toBeTruthy()
+    let startingMode = (await getTraceMode())[0]
+    expect(startingMode).toEqual(MODE.FULL)
+
+    await expect(browser.testHandle.expectResources(3000)).resolves.toBeTruthy()
+  })
+
+  it.withBrowsersMatching(onlyChrome)('does not capture more than the last 30 seconds when error happens', async () => {
+    await getReplayOnErrorUrl.then(builtUrl => browser.url(builtUrl)).then(() => browser.waitForAgentLoad()).then(async () => {
+      await browser.pause(30000)
```

Comment:
can we just make this configurable to prevent needing a test that takes > 30s

Reply:
Yeah, it's a tweak but I could get it out of the way if it's an eventuality.

Reply:
Actually going to ooscope this. We should still test the default behavior, which is 30s, anyways and have another test later for a working redefine-able window.